common/common.cpp

```diff
@@ -1056,7 +1056,7 @@ std::string get_system_info(const gpt_params & params) {
 }
 
 std::string gpt_random_prompt(std::mt19937 & rng) {
-    const int r = rng() % 10;
+    const int r = int(rng() % 10);
```

Comment:
These explicit `int` casts are not necessary - it's redundant information since we clearly have an `int` on the left hand side

Reply:
They are necessary, if you take a look into [MSVC CI log](https://github.com/ggerganov/llama.cpp/actions/runs/7684654611/job/20942846158#step:6:82), you'll notice a bunch of warnings regarding conversions. Each of those warnings is a potential issue with overflow/loss of data.

The most typical fault scenario for this kind of issue is when you have an integer loop counter iterating over a big vector, which will exceed the int_max.